arbos/tx_processor.go

```diff
@@ -132,19 +132,25 @@ func (p *TxProcessor) StartTxHook() (endTxNow bool, gasUsed uint64, err error, r
 
 		// mint funds with the deposit, then charge fees later
 		util.MintBalance(&from, tx.DepositValue, evm, scenario)
+		availableRefund := arbmath.BigSub(tx.DepositValue, tx.Value)
+		availableRefund = arbmath.BigMax(availableRefund, common.Big0)
 
+		// collect the submission fee
 		submissionFee := retryables.RetryableSubmissionFee(len(tx.RetryData), tx.L1BaseFee)
-		excessDeposit := arbmath.BigSub(tx.MaxSubmissionFee, submissionFee)
-		if excessDeposit.Sign() < 0 {
-			return true, 0, errors.New("max submission fee is less than the actual submission fee"), nil
-		}
-
-		// move balance to the relevant parties
 		if err := util.TransferBalance(&from, &networkFeeAccount, submissionFee, evm, scenario); err != nil {
 			return true, 0, err, nil
 		}
-		if err := util.TransferBalance(&from, &tx.FeeRefundAddr, excessDeposit, evm, scenario); err != nil {
-			return true, 0, err, nil
+
+		// refund excess submission fee
+		submissionFeeRefund := arbmath.BigSub(tx.MaxSubmissionFee, submissionFee)
+		if submissionFeeRefund.Sign() < 0 {
+			return true, 0, errors.New("max submission fee is less than the actual submission fee"), nil
+		}
+		submissionFeeRefund = arbmath.BigMin(submissionFeeRefund, availableRefund)
+		submissionFeeRefund = arbmath.BigMin(submissionFeeRefund, statedb.GetBalance(from))
+		availableRefund.Sub(availableRefund, submissionFeeRefund)
```

Comment:
`submissionFee` is spent and `submissionFeeRefund` is refunded
so I think we should subtract by tx.MaxSubmissionFee instead
```suggestion
		availableRefund.Sub(availableRefund, tx.MaxSubmissionFee)
```

Reply:
This code has been changed to charge the submission fee against the max refund first if possible